docs/en/operations/storing-data.md

```diff
@@ -11,4 +11,122 @@ To work with data stored on `Amazon s3` disks use [s3](../engines/table-engines/
+
+Configuration markup:
+
+``` xml
+<yandex>
+    <storage_configuration>
+        <disks>
+            <disk_s3>
+                <type>s3</type>
+                <endpoint>http://minio1:9001/root/data/</endpoint>
+                <access_key_id>minio</access_key_id>
+                <secret_access_key>minio123</secret_access_key>
+            </disk_s3>
+            <disk_memory>
+                <type>memory</type>
+            </disk_memory>
+            <disk_local>
+                <type>local</type>
+                <path>/disk/</path>
+            </disk_local>
+            <disk_s3_encrypted>
+                <type>encrypted</type>
+                <disk>disk_s3</disk>
+                <path>encrypted/</path>
+                <key id="0">firstfirstfirstf</key>
+                <key id="1">secondsecondseco</key>
+                <current_key_id>1</current_key_id>
+            </disk_s3_encrypted>
+            <disk_local_encrypted>
+                <type>encrypted</type>
+                <disk>disk_local</disk>
+                <path>encrypted/</path>
+                <key>abcdefghijklmnop</key>
+            </disk_local_encrypted>
+        </disks>
+    </storage_configuration>
+</yandex>
+```
+
+Required parameters:
+
+-   `type` — `encrypted`. Otherwise, the encrypted disk is not created.
+-   `disk` — Type of disk for data storage.
+-   `key` — The key for encryption and decryption. Type: [Uint64](../sql-reference/data-types/int-uint.md). You can use `key_hex` parameter to encrypt in hexadecimal form.
+    You can specify multiple keys using the `id` attribute (see example).
+
+Optional parameters:
+
+-   `path` — Path to the location on the disk where the data will be saved. If not specified, the data will be saved to the root of the disk.
+-   `current_key_id` — The key is used for encryption, and all the specified keys can be used for decryption. This way, you can switch to another key, while maintaining access to previously encrypted data.
+-   `algorithm` — [Algorithm](../sql-reference/statements/create/table.md#create-query-encryption-codecs) for encryption. Can have one of the following values: `AES_128_CTR`, `AES_192_CTR` or `AES_256_CTR`. By default: `AES_128_CTR`. The key length depends on the algorithm: `AES_128_CTR` — 16 bytes, `AES_192_CTR` — 24 bytes, `AES_256_CTR` — 32 bytes.
+
+Example of disk configuration:
+
+``` xml
+<disks>
+  <disk1>
+    <type>local</type>
+    <path>/path1/</path>
+  </disk1>
+  <disk2>
+    <type>encrypted</type>
+    <disk>disk1</disk>
+    <path>path2/</path>
+    <key>...</key>
+  </disk2>
+</disks>
+```
+
+For example, when ClickHouse writes data from some table in the form of a file `store/all_1_1_0/data.bin` to disk1, then in fact this file will be written to the physical disk along the path `/path1/store/all_1_1_0/data.bin`.
+
+When writing the same file to disk2, it will actually be written to the physical disk at the path `/path1/path2/store/all_1_1_0/data.bin` in encrypted form.
```

Comment:
```suggestion
When writing the same file to `disk2`, it will actually be written to the physical disk at the path `/path1/path2/store/all_1_1_0/data.bin` in encrypted mode.
```